File: frontend/src/api/order.ts
import request from './request'

export interface Order {
  id: number
  orderNo: string
  senderName: string
  senderPhone: string
  senderAddress: string
  senderLng: number
  senderLat: number
  receiverName: string
  receiverPhone: string
  receiverAddress: string
  receiverLng: number
  receiverLat: number
  goodsName: string
  weight: number
  volume: number
  status: string
  remark: string
  creatorId: number
  createdAt: string
  updatedAt: string
}

export interface OrderReviewRequest {
  action: string
  remark?: string
}

export interface OrderQuery {
  page: number
  size: number
  status?: string
  startDate?: string
  endDate?: string
  keyword?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export const orderApi = {
  list(params: OrderQuery) {
    return request.get<unknown, { data: PageResult<Order> }>('/orders', { params })
  },
  listMine(params: OrderQuery) {
    return request.get<unknown, { data: PageResult<Order> }>('/orders/mine', { params })
  },
  getById(id: number) {
    return request.get<unknown, { data: Order }>(`/orders/${id}`)
  },
  create(data: Partial<Order>) {
    return request.post<unknown, { data: Order }>('/orders', data)
  },
  update(id: number, data: Partial<Order>) {
    return request.put<unknown, { data: Order }>(`/orders/${id}`, data)
  },
  delete(id: number) {
    return request.delete(`/orders/${id}`)
  },
  cancel(id: number) {
    return request.patch(`/orders/${id}/cancel`)
  },
  cancelPendingReview(id: number) {
    return request.patch(`/orders/${id}/cancel`)
  },
  review(id: number, action: string, remark?: string) {
    const data: OrderReviewRequest = { action, remark }
    return request.patch(`/orders/${id}/review`, data)
  },
  changeStatus(id: number, status: string, remark?: string) {
    return request.patch(`/orders/${id}/status`, { targetStatus: status, remark })
  },
}

File: backend/order-service/src/main/java/com/smart/order/entity/Order.java
package com.smart.order.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.smart.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("orders")
public class Order extends BaseEntity {
    private String orderNo;
    private String senderName;
    private String senderPhone;
    private String senderAddress;
    private BigDecimal senderLng;
    private BigDecimal senderLat;
    private String receiverName;
    private String receiverPhone;
    private String receiverAddress;
    private BigDecimal receiverLng;
    private BigDecimal receiverLat;
    private String goodsName;
    private BigDecimal weight;
    private BigDecimal volume;
    /** pending_review / pending / dispatched / in_progress / completed / cancelled / exception */
    private String status;
    private String remark;
    private Long creatorId;
}
